test/y2storage/filesystems/btrfs_test.rb

```diff
@@ -194,25 +194,114 @@
   describe "#create_btrfs_subvolume" do
     let(:devicegraph) { Y2Storage::StorageManager.instance.staging }
 
-    let(:path) { "@/foo" }
+    let(:path1) { "@/foo" }
+    let(:path2) { "@/foo/bar/baz" }
+    let(:path3) { "@/foo/bar" }
     let(:nocow) { true }
 
     it "creates a new subvolume" do
-      expect(filesystem.btrfs_subvolumes.map(&:path)).to_not include(path)
-      filesystem.create_btrfs_subvolume(path, nocow)
-      expect(filesystem.btrfs_subvolumes.map(&:path)).to include(path)
+      expect(filesystem.btrfs_subvolumes.map(&:path)).to_not include(path1)
+      filesystem.create_btrfs_subvolume(path1, nocow)
+      expect(filesystem.btrfs_subvolumes.map(&:path)).to include(path1)
     end
 
     it "returns the new created subvolume" do
-      subvolume = filesystem.create_btrfs_subvolume(path, nocow)
+      subvolume = filesystem.create_btrfs_subvolume(path1, nocow)
       expect(subvolume).to be_a(Y2Storage::BtrfsSubvolume)
-      expect(subvolume.path).to eq(path)
+      expect(subvolume.path).to eq(path1)
       expect(subvolume.nocow?).to eq(nocow)
     end
 
     it "creates the subvolume with the correct mount point" do
-      subvolume = filesystem.create_btrfs_subvolume(path, nocow)
-      expect(subvolume.mount_path).to eq("/foo")
+      subvolume = filesystem.create_btrfs_subvolume(path1, nocow)
+      expect(subvolume.mount_path).to eq(path1.delete("@"))
+    end
+
+    context "when the filesystem is going to be formatted" do
+      before do
+        allow(filesystem).to receive(:exists_in_probed?) .and_return(false)
+      end
+
+      it "can insert a subvolume into an existing hierarchy" do
+        filesystem.create_btrfs_subvolume(path2, nocow)
+        filesystem.create_btrfs_subvolume(path3, nocow)
+        expect(filesystem.btrfs_subvolumes.map(&:path)).to include(path3)
+      end
+    end
+
+    context "when the filesystem already exists" do
```

Comment:
It also makes sense to have another test checking that it returns nil if the subvolume couldn't be inserted

Reply:
This is probably what you wanted to express:

```ruby
context "when the filesystem already exists and the subvolume does not fit in the existing hierarchy" do
  before do
    allow(filesystem).to receive(:exists_in_probed?) .and_return(true)
    filesystem.create_btrfs_subvolume(path2, nocow)
  end

  it "does not create the subvolume" do
    filesystem.create_btrfs_subvolume(path3, nocow)
    expect(filesystem.btrfs_subvolumes.map(&:path)).to_not include(path3)
  end

  it "returns nil" do
    result filesystem.create_btrfs_subvolume(path3, nocow)
    expect(result).to be_nil
  end
end
```